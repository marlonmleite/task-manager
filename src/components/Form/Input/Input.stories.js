import React from 'react'
import { storiesOf } from '@storybook/react'
import { InputComponent as Input } from './'

const stories = storiesOf('Input', module)

stories.add('default', () => (
  <Input defaultValue="teste" />
))

stories.add('disabled', () => (
  <Input defaultValue="teste" disabled />
))
