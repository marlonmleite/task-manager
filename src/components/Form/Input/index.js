import styled from 'styled-components'
import withFormGroup from '../withFormGroup'

export const InputComponent = styled.input`
  width: 100%;
  font-size: 15px;
  padding: 5px 5px 5px 0;
  height: 24px;
  border-color: #e5e5e5;
  border-style: solid;
  border-width: 0 0 1px 0;
  border-radius: 0;
  color: #222;
  line-height: 24px;
  background-color: transparent;
  box-sizing: content-box;
 
  :focus {
    outline: none;
    border-color: ${({ theme }) => theme.focus};
  }

  :disabled {
    cursor: not-allowed;
  }
`

InputComponent.defaultProps = {
  type: 'text',
}

export const Input = withFormGroup(InputComponent)
