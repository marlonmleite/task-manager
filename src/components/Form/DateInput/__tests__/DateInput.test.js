import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import DatePicker from 'react-datepicker'
import { InputComponent } from '../../Input'
import { DateInputComponent } from '../'

describe('DateInput', () => {
  it('should render', () => {
    const input = shallow(<DateInputComponent name="a" onChange={jest.fn()} />)

    expect(input).toMatchSnapshot()
  })

  it('should display date picker', () => {
    const input = shallow(<DateInputComponent name="a" onChange={jest.fn()} />)
    const event = { preventDefault: jest.fn() }

    input.find(InputComponent).simulate('focus', event)

    const datePicker = input.find(DatePicker)

    expect(event.preventDefault).toHaveBeenCalled()
    expect(datePicker.exists()).toBeTruthy()
  })

  it('should handle onChange', () => {
    const onChange = jest.fn()
    const input = shallow(<DateInputComponent name="a" onChange={onChange} />)

    input.find(InputComponent).simulate('focus')
    input.find(DatePicker).simulate('change', moment())

    expect(onChange).toHaveBeenCalledWith('1970-01-01T00:00:00.000Z')
  })
})
