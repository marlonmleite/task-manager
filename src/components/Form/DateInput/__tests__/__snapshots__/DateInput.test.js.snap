// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput should render 1`] = `
<Fragment>
  <styled.input
    onFocus={[Function]}
    type="text"
    value=""
  />
</Fragment>
`;
